https://github.com/bitcoin/bitcoin/actions/runs/12711073565/job/35433597593?pr=31635#step:7:44312:
```bash
 test  2025-01-10T14:30:30.975000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/home/runner/work/_temp/test/functional/test_framework/util.py", line 160, in try_rpc
                                       fun(*args, **kwds)
                                     File "/home/runner/work/_temp/test/functional/test_framework/coverage.py", line 50, in __call__
                                       return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                     File "/home/runner/work/_temp/test/functional/test_framework/authproxy.py", line 146, in __call__
                                       raise JSONRPCException(response['error'], status)
                                   test_framework.authproxy.JSONRPCException: createwalletdescriptor is not available for non-descriptor wallets (-4)
                                   During handling of the above exception, another exception occurred:
                                   Traceback (most recent call last):
                                     File "/home/runner/work/_temp/test/functional/test_framework/test_framework.py", line 135, in main
                                       self.run_test()
                                     File "/home/runner/work/_temp/ci/scratch/build-x86_64-pc-linux-gnu/test/functional/wallet_address_types.py", line 372, in run_test
                                       assert_raises_rpc_error(-5, "Unknown address type 'bech23'", self.nodes[3].createwalletdescriptor, "bech23")
                                     File "/home/runner/work/_temp/test/functional/test_framework/util.py", line 151, in assert_raises_rpc_error
                                       assert try_rpc(code, message, fun, *args, **kwds), "No exception raised"
                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                     File "/home/runner/work/_temp/test/functional/test_framework/util.py", line 164, in try_rpc
                                       raise AssertionError("Unexpected JSONRPC error code %i" % e.error["code"])
                                   AssertionError: Unexpected JSONRPC error code -4
 test  2025-01-10T14:30:31.007000Z TestFramework (DEBUG): Closing down network thread
```

--- a/test/functional/wallet_address_types.py
+++ b/test/functional/wallet_address_types.py
@@ -369,6 +369,8 @@ class AddressTypeTest(BitcoinTestFramework):
         assert_raises_rpc_error(-5, "Unknown address type ''", self.nodes[3].getnewaddress, None, '')
         assert_raises_rpc_error(-5, "Unknown address type ''", self.nodes[3].getrawchangeaddress, '')
         assert_raises_rpc_error(-5, "Unknown address type 'bech23'", self.nodes[3].getrawchangeaddress, 'bech23')
+        if self.options.descriptors:
+            assert_raises_rpc_error(-5, "Unknown address type 'bech23'", self.nodes[3].createwalletdescriptor, "bech23")
 
         self.log.info("Nodes with changetype=p2sh-segwit never use a P2WPKH change output")
         self.test_change_output_type(4, [to_address_bech32_1], 'p2sh-segwit')

